lib/environment/i18n.js

```diff
@@ -26,9 +26,14 @@ export async function _loadI18n(): Promise<void> {
 	// will be hit the first (ever) pageload on a new domain
 	// or after clearing localStorage
 	messages = await sendMessage('i18n', rawLocale());
-	localStorage.setItem(CACHED_MESSAGES_KEY, JSON.stringify(messages));
-	localStorage.setItem(CACHED_LANG_KEY, rawLocale());
-	localStorage.setItem(CACHED_MESSAGES_TOKEN_KEY, buildToken);
+
+	try { // Fails if remaining localStorage space is insufficient
+		localStorage.setItem(CACHED_MESSAGES_KEY, JSON.stringify(messages));
+		localStorage.setItem(CACHED_LANG_KEY, rawLocale());
+		localStorage.setItem(CACHED_MESSAGES_TOKEN_KEY, buildToken);
+	} catch (e) {
+		console.error('Could not cache i18n', e);
```

Comment:
Hmm I don't really like this as-is because it'll make RES silently run slowly.
IMO we might as well fix it for the case we know about, even if it's a bit hacky.
(When the settings console is moved to a separate page we can probably split the i18n files, inline the non-option-related translations and get rid of this entirely)

I'm thinking:
- find the largest localStorage key
- if it's a known key (`ads.adserverDownvotePixel`, `ads.adserverUpvotePixel`, `ads.recent`) delete it
(or maybe just delete those keys unconditionally in this catch block, I doubt any users will complain about us deleting reddit's ad tracking)